Gitignore, javafiles, read in java codes

mccurdy42 opened the pull request:
Changes
- codes to read in the first excel sheet. modified some of the arrays to arraylists so be aware that the syntax needs to change to access (tiny change though)

- added a .gitignore to limit additional junk on git (like the .metadata files that eclipse generates). If there are additional problems or more files that you would like to add, https://www.gitignore.io/, is a good resource

Testing  instructions: 
- Modify library paths (cplex and poi paths to match your directory) 
- Modify excel read in path
- Add in near the bottom of file 
`           System.out.println("N2 "+n2);
           System.out.println("Schedulue Name " + schedule_name);
           System.out.println("teacher name" +teacherNames);
           System.out.println("tacher allocation " +FTE);
           for ( int p = 0; p< availableTime.size();p++) {
        	   	System.out.println("teacher time slot" + Arrays.toString(availableTime.get(p)) );
           }
           System.out.println("french teach lb" +frenchTeachlb);
           System.out.println("fremch num" + frenchNum);`

Expected output with the attached excel sheet should be:

> N2 4
Schedulue Name SAMPLE TITLE 
teacher name[Example full-time Teacher, Example part-time Teacher, teacher a, teacher b]
tacher allocation [1.0, 0.5, 1.0, 1.0]
teacher time slot[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
teacher time slot[1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0]
teacher time slot[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
teacher time slot[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
french teach lb1
fremch num8

mccurdy42 committed:
Gitignore, javafiles, read in java codes

ahopkins19 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -1,14 +1,181 @@
+        //index of first french teacher 
+        int frenchTeachlb = -1;
+        //change to first sheet numberOfSheets to 1
+        for (int i = 0; i < 1; i++) {
```
does this need to be written as a loop if its just going once?

mestrona3 replied:
I think we should also change the name of this variable so its more clear

mccurdy42 replied:
My assumption that this would loop over each sheet. However, I can change it if other people are access their sheets alternatively

ahopkins19 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -1,14 +1,181 @@
+            //index of start of teacher matrix
+            int teacher_matrix_start = 6;
+            //teacher name col
+            int teacher_name_row = 0;
```
can you change these variable names to col? I'm pretty sure they are referencing columns not rows

mestrona3 reviewed (changes requested):
Names and time slots were imported great, I tested those and had no issues! Please see the rest of the comments on the PR to make those changes though. Teacher name array needs to exclude the examples (the first two rows), as does teacher allocation, and then we don't need to print teacher time slot for these. Also - french teach, could we change this to a different more clear variable name? Also, n3 is still hardcoded and isn't based on the right sheet.

ahopkins19 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -1338,7 +1508,16 @@ else if(i==awaySubject) {
 else {
 	System.out.println("Model not solved");
 	cplex.exportModel("lpex1.lp");
+}*/
```
make sure this doesnt get pushed

mestrona3 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -17,12 +184,14 @@ public static void modelConfig() {
+//		String [] teacherNames;
 	
 	//define parameters - cohorts
 		int n3 = 13;
```
This needs to be variable, its messing up printing 'frenchnum', also confused about the difference between frenchnum and frenchteachlb as all we need to do is collect the number of french teachers, but this could just be a point of clarification on my end

mccurdy42 replied:
french num is dependent on N3 which is taken in on a different page and should be dynamically updated once we have page two and french teachers are listed at the end of the teacher list

mestrona3 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -1,14 +1,181 @@
+            		//interate over two rows at a time,
+            		//then do a for loop for each two
+            		int lastColumn =  Math.max(currRow.getLastCellNum(), 10);
+            		for (int k = 0; k < lastColumn; k++) {
```
this should start after the 2 examples rows

ahopkins19 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -1,14 +1,181 @@
+            			 		} 	
+            			 }
+            		}
+            		int day1 = 4;
```
rename these variables (day1 and day5) for more clarity.. day1_col  or something

ahopkins19 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -1,14 +1,181 @@
+            		int day5 = 8;
+            		if (fullTime == false) {
+            			int[] allocation_times = new int[30]; 
+            			int a_pointer = 0;
```
what is a_pointer? maybe rename

ahopkins19 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -17,12 +184,14 @@ public static void modelConfig() {
+//		double [] FTE = {1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,0.2,0.2,0.6,0.2,1.0,1.0};
+//		int frenchTeachlb = n2-2;
 		int frenchTeachub = n2-1;
 		int frenchTeach = 2;
```
this variables is used for indeces in the model and shouldnt be removed yet

mccurdy42 committed:
changes

mccurdy42 committed:
stuck with frenchlb on other

mccurdy42 committed:
resolve french

mccurdy42 commented:
removed french num variable so the print statement for french num should be removed. 
I also changed the input excel file to match the user expectation that french teachers should be listed

mestrona3 reviewed (approved):
I'm good with the functionality that this PR provides. It brings in what we need - schedule title, teacher names, their FTE and availability.  I still have slight confusion around FrenchTeachLB but I think that can be resolved by explaining it in our meeting tomorrow, and am happy with this being merged. 

Please don't forget to uncomment solver before merging!

mestrona3 commented on FYDP_ModV1/src/version1.java:
```diff
@@ -1,14 +1,181 @@
+
+            }
+           n2 = teacherNames.size();
+           int frenchTeach = n2 - first_french_teacher;
```
I think this should change to 'int frenchTeach = n2 - first_french_teacher + 1; otherwise you exclude that 'first' french teacher from the count.... however, this only works if there is at least one french certified teacher. We can either assume there is always one french certified teacher or add in an if statement to calculate this variable based on if the 'french certification' column is populated

mccurdy42 committed:
fixing french
